base_layer/core/src/base_node/comms_interface/inbound_handlers.rs

```diff
@@ -506,7 +503,35 @@ where B: BlockchainBackend + 'static
         Ok(())
     }
 
-    async fn check_min_block_difficulty(&self, new_block: &NewBlock) -> Result<(), CommsInterfaceError> {
+    /// Handles a `NewBlock` message. Only a single `NewBlock` message can be handled at once to prevent extraneous
+    /// requests for the full block.
+    /// This may (asynchronously) block until the other request(s) complete or time out and so should typically be
+    /// executed in a dedicated task.
+    pub async fn handle_new_block_message(
+        &mut self,
+        new_block: NewBlock,
+        source_peer: NodeId,
+    ) -> Result<(), CommsInterfaceError> {
+        if let Err(err @ CommsInterfaceError::ChainStorageError(ChainStorageError::ValidationError { .. })) =
```

Comment:
You are swallowing all the other error types here.... I think this is worse

Reply:
your right,this is worse